python/cudf/cudf/core/column/column.py

```diff
@@ -3289,7 +3289,7 @@ def deserialize_columns(headers: list[dict], frames: list) -> list[ColumnBase]:
     return columns
 
 
-def concat_columns(objs: "MutableSequence[ColumnBase]") -> ColumnBase:
+def concat_columns(objs: Sequence[ColumnBase]) -> ColumnBase:
```

Comment:
OK, so this change is so that `concat_columns` doesn't both give you a concatenated output _and_ mutate your input list?

That seems reasonable I think.

The only thing I can think might be problematic is if `objs` is the only reference to each column is in this list, previously we used slightly less memory (because we overwrite the all null columns with differing dtypes).

But overall I much prefer this implementation.